src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -1431,11 +1453,108 @@ static ulong ReadVarUInt64(ref byte* pCursor)
             return val;
         }
 
-        public static void ReadFromFormatV4(byte* headerPtr, bool useHeaderCompresion, ref EventPipeEventHeader header)
+        public static void ReadFromFormat(int version, byte* headerPtr, bool useHeaderCompresion, ref EventPipeEventHeader header)
+        {
+            switch (version)
+            {
```

Comment:
I'm a bit wary of mixing v4 and v5 functionality and having a single implementation for both.  I realize this might make for a little code duplication. Presumably any feature work we do in the runtime that would also necessitate a version increase will get rolled into v5 as well.  This could mean that we need to bring back the v4 version of the code later anyway if the delta becomes large enough.